source/io/module_javaScriptInvoke.js

```diff
@@ -117,12 +97,12 @@ var assignJavaScriptInvoke;
         };
 
         Module.javaScriptInvoke.readJavaScriptRefNum = function (javaScriptValueRef) {
-            var cookie = Data_ReadJavaScriptRefNum(javaScriptValueRef.dataRef);
+            var cookie = Module.eggShell.readDouble(javaScriptValueRef);
```

Comment:
Using readDouble to deal with refnum does not feel right at the API level. It is probably an implementation detail that readDouble works for refnum. OTOH, the API may be generalized for "Refnum" and not specific to "JavascriptRefNum" if that makes sense.

Reply:
Nope, disagree. The major idea of the new api is to focus on reading values out of Vireo's memory as the host memory type. Previously we had hundreds of functions for converting each vireo type to a type supported in JS. The new api flips it on it's head. There are a handful of functions that can be used to can handle lots of vireo types.

The previous function read a JavaScriptRefNum type and resulted in a JavaScript number as a double. That IS what it did just with tons of implicit conversions. If reading as a double felt wrong, too late that is what we were doing, it just wasn't explicit.

This makes it very explicit.

Now the ReadDoubleFromMemory function implements very specifically how the Refnum is represented as a double: https://github.com/ni/VireoSDK/pull/523/files#diff-f2734a9037270e6cd627741ed25d331aR2724

So instead of hiding behind implicit conversions from Int32 to double, this codifies it explicitly.

Reply:
This discussion did make me realize that if the definition of refnum changed from UInt32 to UInt64 then this code should fail as a UInt64 RefNum cannot be safely represented as a double (other UInts will loose precision which is a choice the user can make but that is not a valid choice for RefNums).

So I made the following change in this PR:
https://github.com/ni/VireoSDK/pull/523/files#diff-f2734a9037270e6cd627741ed25d331aR2722

https://github.com/ni/VireoSDK/pull/523/files#diff-f2734a9037270e6cd627741ed25d331aR2792